src/storage/file_storage.rs

```diff
@@ -0,0 +1,274 @@
+use std::{
+    collections::{HashMap, VecDeque},
+    fs::{self, File},
+    io::{self, BufRead},
+    path::PathBuf,
+    time::SystemTime,
+};
+
+use crate::Error;
+
+pub struct FileWriterFactory {
```

Comment:
Please no Factorys or Interactors. Just write functions that do what you intend and logically stand on their own

Example:

```rust
struct Directory {
  path: PathBuf
}

impl Directory {
pub fn get_file_names(&self) -> Result<Vec<String>, Error> {}
pub fn get_most_recent(&self) -> Result<Option<String>, Error> {}
}

struct File { 
  path: PathBuf
}

// although I'm not sure if this is even necessary since there are std lib functions for this
impl File {
  pub fn read
  pub fn write
}

File::new(DataPaths::Log.path().join("log_file.txt")).write("abc")
```